### DISCUSSION TITLE
A few questions regarding acceptable return values

### DISCUSSION THREAD

**Discussion by axvr** [PlummersSoftwareLLC/Primes#794] [Q&A] [answered]
Hi I have a few of questions regarding return values.

1. The [rules](https://github.com/PlummersSoftwareLLC/Primes/blob/drag-race/CONTRIBUTING.md#rules) state you can return "either a list of primes or the `is_prime` array, containing the result of the sieve".  Am I correct in thinking that the `is_prime` array can just be an array of bits, where the index of the bit corresponds to the value?  (I.e. we don't have to return a proper list of only prime numbers?)

2. If the above is allowed: is returning an `is_array` value without the multiples of 2 allowed?  (I.e. an array of half the size where each bit corresponds to double of the index.)

3. Does the sieve result need validating on every pass or can validation be deferred until after the 5 seconds are up?

**Comment by rbergen** [comment, marked as answer]
1. In the context of the rules, the `is_prime` array has come to be interpreted as "whatever the memory map is for the sieve". That can indeed be a situation where the index of bits represent the prime candidates. In that, the bits may also be inverted if that is how the specific implementation computes the primes.

2. Yes. In line with the previous answer, that is allowed.

3. In principle, validating the sieve result is not a strict requirement at all from the perspective of the rules. It is a very wise step to take for obvious reasons, but can then be done once and outside of the timed loop.
